packages/in_app_purchase/in_app_purchase_storekit/test/fakes/fake_storekit_platform.dart

```diff
@@ -232,6 +232,66 @@ class FakeStoreKitPlatform {
   Map<String, Object?> _getArgumentDictionary(MethodCall call) {
     return (call.arguments as Map<Object?, Object?>).cast<String, Object?>();
   }
+
+  @override
+  bool canMakePayments() {
+    return true;
+  }
+
+  @override
+  void addPayment(Map<String?, Object?> paymentMap) {
+    final String id = paymentMap['productIdentifier']! as String;
+    final int quantity = paymentMap['quantity']! as int;
+
+    // Keep the received paymentDiscount parameter when testing payment with discount.
+    if (paymentMap['applicationUsername']! == 'userWithDiscount') {
+      final Map<dynamic, dynamic>? discountArgument =
+          paymentMap['paymentDiscount'] as Map<dynamic, dynamic>?;
+      if (discountArgument != null) {
+        discountReceived = discountArgument.cast<String, dynamic>();
+      } else {
+        discountReceived = <String, dynamic>{};
+      }
+    }
+
+    final SKPaymentTransactionWrapper transaction =
+        createPendingTransaction(id, quantity: quantity);
+    transactionList.add(transaction);
+    InAppPurchaseStoreKitPlatform.observer.updatedTransactions(
+        transactions: <SKPaymentTransactionWrapper>[transaction]);
+    sleep(const Duration(milliseconds: 30));
+    if (testTransactionFail) {
+      final SKPaymentTransactionWrapper transactionFailed =
+          createFailedTransaction(id, quantity: quantity);
+      InAppPurchaseStoreKitPlatform.observer.updatedTransactions(
+          transactions: <SKPaymentTransactionWrapper>[transactionFailed]);
+    } else if (testTransactionCancel > 0) {
+      final SKPaymentTransactionWrapper transactionCanceled =
+          createCanceledTransaction(id, testTransactionCancel,
+              quantity: quantity);
+      InAppPurchaseStoreKitPlatform.observer.updatedTransactions(
+          transactions: <SKPaymentTransactionWrapper>[transactionCanceled]);
+    } else {
+      final SKPaymentTransactionWrapper transactionFinished =
+          createPurchasedTransaction(
+              id, transaction.transactionIdentifier ?? '',
+              quantity: quantity);
+      InAppPurchaseStoreKitPlatform.observer.updatedTransactions(
+          transactions: <SKPaymentTransactionWrapper>[transactionFinished]);
+    }
+  }
+
+  @override
+  SKStorefrontMessage storefront() {
+    // TODO(louisehsu): implement storefront
+    throw UnimplementedError();
+  }
+
+  @override
+  List<SKPaymentTransactionMessage?> transactions() {
+    // TODO(louisehsu): implement transactions
```

Comment:
These are left to do outside of this PR, right?  You will have to write up issues and link them instead of `louisehsu`.

Reply:
even if theyre only being used for testing? 👀

Reply:
Yea, if it's something you think we should do, have it be something like `// TODO(http://github/flutter/issues/1234): implement these`.  If you don't think it's something we need to come back to, just remove the TODO.